tests/lib/rules/no-var-requires.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @fileoverview Disallows the use of require statements except in import statements.
+ * @author Macklin Underdown
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/no-var-requires"),
+    RuleTester = require("eslint").RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+    parser: "typescript-eslint-parser"
+});
+
+ruleTester.run("no-var-requires", rule, {
+    valid: ["import foo = require('foo')"],
+    invalid: [
+        {
+            code: "var foo = require('foo')",
+            errors: [
+                {
+                    message: "Require statement not part of import statement",
+                    line: 1,
+                    column: 11
+                }
+            ]
+        },
+        {
+            code: "const foo = require('foo')",
+            errors: [
+                {
+                    message: "Require statement not part of import statement",
+                    line: 1,
+                    column: 13
+                }
+            ]
+        },
+        {
+            code: "let foo = require('foo')",
+            errors: [
+                {
+                    message: "Require statement not part of import statement",
+                    line: 1,
+                    column: 11
+                }
+            ]
+        },
+        {
+            code: "require('foo')",
```

Comment:
I believe an unassigned call to `require()` is invalid, but I'm not 100% sure about this based on TSLint's [`no-var-requires` tests](https://github.com/palantir/tslint/blob/master/test/rules/no-var-requires/test.ts.lint). Just calling this out in case anyone has more experience with TSLint's `no-var-requires` rule and if this truly an invalid case.

Reply:
You can replace `require("foo");` with `import "foo";`. Hence an error on this require call is expected.

Reply:
Forgot about that, thanks for pointing that out. I’ll push up a commit next week when I get access to a computer